src/aerugo.rs

```diff
@@ -346,7 +368,15 @@ impl RuntimeApi for Aerugo {
     type Duration = crate::time::TimerDurationU64<1_000_000>;
 
     fn get_system_time(&'static self) -> Self::Instant {
-        self.hal.get_system_time()
+        // SAFETY: This is safe, because it's a single-core environment,
+        // and no other references to Hal should exist during this call.
+        unsafe {
+            self.hal
+                .as_ref()
+                .as_ref()
```

Comment:
Double `as_ref`?

Reply:
Yes, first borrows from cell and second from Option.

Reply:
ack